apps/emqx_prometheus/src/emqx_prometheus.erl

```diff
@@ -114,16 +113,12 @@ handle_info(_Msg, State) ->
 
 push_to_push_gateway(Uri, Headers, JobName) when is_list(Headers) ->
     [Name, Ip] = string:tokens(atom_to_list(node()), "@"),
-    JobName1 = emqx_placeholder:preproc_tmpl(JobName),
-    JobName2 = binary_to_list(
-        emqx_placeholder:proc_tmpl(
-            JobName1,
-            #{<<"name">> => Name, <<"host">> => Ip}
-        )
+    JobName1 = emqx_connector_template:render_strict(
```

Comment:
Won't this also break compatibility?

Reply:
You're right. Made sure to keep rough backwards compat in d05639fa7e8752ba7ffe25ed4669b4430e9337e1: rendering errors won't result in process exiting. Unsure if we need to go as far as keeping rendering missing bindings as `"undefined"`.

Reply:
We should perhaps keep this behavior (inherited from v4).
It's probably used as a feature in EMQX's downstream.

Reply:
We may in the future enhance with:
`${var}` equivalent to `${var | default('undefined')}`
`${var | default('foo')}` % defaults to 'foo'
`${var!}` % raise exception (cause error for rule-engine and discard this data)

Reply:
Reverted to this behavior in [745fb0d](https://github.com/emqx/emqx/pull/10442/commits/745fb0dca7b25d9c9b809adb3125e8e9a55d677a).